File: src/server/dtos/lecturer.dto.js
import { majorDto } from './major.dto.js';
import { userDto } from './user.dto.js';

const lecturerDto = {
    id: true,
    lecturer_id: true,
    major: {
        select: majorDto,
    },
    user: {
        select : userDto,
    }
};

export { lecturerDto };

File: src/server/AspNet/Application/Common/DTOs/LecturerDto.cs
namespace Application.Common.DTOs
{
    public class LecturerDto
    {
        public string cardId { get; set; } = null!;
        public string status { get; set; } = null!;
        public string position { get; set; } = null!;
        public DateTime joinedAt { get; set; }
        public UserDto user { get; set; } = null!;
        public FacultyDto faculty { get; set; } = null!;
        public MajorDto major { get; set; } = null!;
    }
}